Lib/hmac.py

```diff
@@ -241,13 +241,20 @@ def digest(key, msg, digest):
     if _hashopenssl and isinstance(digest, (str, _functype)):
         try:
             return _hashopenssl.hmac_digest(key, msg, digest)
+        except OverflowError as exc:
+            if (not _hmac and isinstance(digest, str)) or callable(digest):
```

Comment:
I still need to convince myself that this is the correct way to do it. I'm too tired to think.